### ISSUE TITLE
Rust does not warn/error when deriving unknown traits

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang/rust#32349]
`hi.rs`:

```rust
#![feature(custom_derive)]

#[derive(PartialEqq)]
struct A(f32);

fn main() {}
```

```
hi.rs:4:1: 4:15 warning: struct is never used: `A`, #[warn(dead_code)] on by default
hi.rs:4 struct A(f32);
        ^~~~~~~~~~~~~~
hi.rs:3:10: 3:20 warning: unused attribute, #[warn(unused_attributes)] on by default
hi.rs:3 #[derive(PartialEqq)]
                 ^~~~~~~~~~
```

**Comment by frewsxcv** [comment]
I don't know what I was thinking when I opened this since it says `#[derive(PartialEqq)]`
